main.go

```diff
@@ -110,6 +112,24 @@ func mainWithExitCode() exitCode {
 		return code
 	}
 
+	if args.enableGoRuntimeProfiler {
+		addr, _ := strings.CutPrefix(args.agentURL, "http://")
+		err = profiler.Start(
+			profiler.WithService(args.serviceName+"-profiler"),
+			profiler.WithEnv(args.environment),
+			profiler.WithVersion(versionInfo.Version),
+			profiler.WithAgentAddr(addr),
```

Comment:
Profile 1/100 of mutex contention events, which still provides observability, with very low overhead even if the app is very contended.

```suggestion
			profiler.WithAgentAddr(addr),
			profiler.MutexProfileFraction(100),
```